File: employee_skill_tracker/app/models/metric.rb
class Metric < ApplicationRecord
  belongs_to :user
  belongs_to :employee
  
  validates :name, presence: true
  validates :value, presence: true, numericality: true
  validates :timestamp, presence: true
  validates :level, presence: true, inclusion: { 
    in: ['Beginner', 'Intermediate', 'Advanced', 'Expert'],
    message: "%{value} is not a valid level" 
  }
  
  scope :recent, -> { order(timestamp: :desc) }
  scope :by_date_range, ->(start_date, end_date) { 
    where(timestamp: start_date..end_date) if start_date && end_date 
  }
end

File: employee-skill-frontend/src/types/metric.ts
export type SkillLevel = "Beginner" | "Intermediate" | "Advanced" | "Expert";

export interface Metric {
  id: number;
  name: string;
  value: number;
  timestamp: string;
  level: SkillLevel;
  evidence?: string;
  employee: {
    id: number;
    name: string;
  };
}

export const SKILL_LEVELS: SkillLevel[] = [
  "Beginner",
  "Intermediate",
  "Advanced",
  "Expert",
];
export const METRIC_NAMES = [
  "Performance",
  "Productivity",
  "Quality",
  "Communication",
];
